docs/docsite/rst/network_declarative_intent.rst

```diff
@@ -0,0 +1,360 @@
+*********************************
+Declarative Intent in Networking
+*********************************
+
+.. contents:: Topics
+
+This section explores what `Declarative Intent modules` are within the context of Ansible Networking.
+
+This document is part of a collection on Networking. The complete list of guides can be found at :ref:`Network Guides <guide_networking>`.
+
+Overview
+========
+
+INTERNAL NOTE: TBD - Short description of the problem. Needs to be very simple
+
+State what you want the end result to be, not how go get there
+This results in easier to understand playbooks, which internally deal with the differences between Network OS versions.
+
+
+Reminder: Ansible isn't a monitoring tool.
+
+
+
+Without declarative intent
+==========================
+
+INTERNAL NOTE: TBD - Simplist case that shows the issue. We can build on this later
+
+Enable a port and set vlan
+
+.. code-block:: yaml
+
+   - name: Enable port
+
+Ansible reports everything is OK.
+
+The issue here is that we are only asserting *configuration* changes, there is no checks of the *state* of the network infrastructure, for example has the cable been plugged in?
+
+
+With declaritive intent
+=======================
+
+INTERNAL NOTE: TBD - Simplist case that shows the solution. We can build out the example in the later use-cases.
+
+**Simple example**
+
+.. code-block:: yaml
+
+   - name: configure interface
+     net_interface:
+       # Declared configuration
+       name: GigabitEthernet0/2
+       description: public interface configuration
+       enabled: yes
+
+       # Intended state
+       state: connected
+
+**Complex example**
+
+.. code-block:: yaml
+
+   - name: configure interface
+     net_interface:
+       # Declared configuration
+       description: public interface configuration
+       enabled: yes
+       aggregate:
+         - GigabitEthernet0/2
+         - GigabitEthernet0/3
+         - GigabitEthernet0/4
+
+       # Intended state
+       state: connected
+       txrate: 10gb
+
+FIXME: Check syntax
+
+
+
+Declarative intent - the details
+================================
+
+INTERNAL NOTE: Only now that we've explained the problem and given an example should we go into details.
+
+
+Declarative intent modules are designed to provide playbook designers a set of network modules that perform declarative configuration tasks on network devices.  This includes the ability to declaratively describe a configuration set.  In addition, declarative intent modules will also provide a means for declaratively expressing the intended ephemeral state of configuration resources.
```

Comment:
Add some context: 

Previous network modules allowed users to update a device by listing out the steps that need to be taking in order to achieve a desired state. Declarative intent modules......